### PR TITLE
Fix styling bugs on mobile - REVIEW ASAP

### PR DESCRIPTION
**Pull request by jacksonrya**
Modal and Button components break the app on mobile.

This fixes that while improving Button component design and providing a better custom scheme provider.

Reviewer: please test on devices.

P.S. The PR template didn't pre-populate this PR. I'm not sure why.

Usage
---
Buttons require a `ButtonStyle` which describes the set of colors used for different states of the button. Each `ButtonStyle` button state accepts a `ColorPalette` object, which currently only describes background and foreground colors.
```tsx
const [ isModalOpen, setIsModalOpen ] = useState(false);

const buttonStyle: ButtonStyle = {
      default: {
         background: 'pink',
         foreground: 'blue',
      },
      pressed: {
         background: 'blue',
         foreground: 'pink',
      },
      disabled: {
         background: 'grey',
         foreground: 'white',
      },
};
```
Button components are a passed a render function. This enables the parent `Button` component to pass the dynamic foreground color to an unknown child. Though, ideally use of `Button` is restricted to be within other Button-type components. So use of any button on a screen is more similar to `TextButton` usage.

`Button` can receive props that determine if it is outlined, as well as all props of `TouchableHighlight`. It can receive a style prop in order to override any provided `ButtonStyle`.
**If an `onPress` callback function is not provided to a `Button`, the `Button` will _look_ pressable but is in fact not--this is intentional in order to increase flexibility and to pass that responsibility to the developer (user of this component); if this is a misleading side-effect, we can force disabled styling as well to visually communicate that the `Button` doesn't do anything without an `onPress` callback.
```tsx
<Button
   buttonStyle={buttonStyle}
   onPress={() => setIsModalOpen(true)}
>
   {
      foregroundColor => (
         <Text style={{ color: foregroundColor }}>
            Open Modal
         </Text>
      )
   }
</Button>
```
Modal accepts a `ColorPalette` which determines the header colors.
```tsx
<Modal
   title="Title in Header"
   open={isModalOpen}
   onDismiss={() => console.log('user tapped behind modal')}
   palette="accent"
>
   {/* Content in the body of the modal */}
   <View>
      <TextButton
         text="Close modal"
         onPress={() => setIsModalOpen(false)}
         buttonStyle={buttonStyle}
      />
   </View>
</Modal>
```
---
> modal-button-preview
_Example Code Preview -- The 'dismiss message' logic is not present in example code_

### CODE REVIEW

File: src/elements/Button/index.types.ts [hunk 1]
```diff
@@ -0,0 +1,7 @@
+import { ColorPalette } from '@util/colorSchemes';
+
+export type ButtonStyle = {
```
Comment (duskcloudxu) [inline src/elements/Button/index.types.ts:3]: Confused about that interface...also it takes much time to find how to use button component, maybe we should add a section of usage example in PR when the PR is about a new component?

Comment (jacksonrya) [reply src/elements/Button/index.types.ts:3]: Apologies... I think that's a good idea. I'll add a usage example in the PR write-up

Comment (jacksonrya) [reply src/elements/Button/index.types.ts:3]: Example and write-up added! I've also pushed a commit to fix that closed modal styling mistake.

**Comment by jacksonrya** [comment]
I think more research into Bit is a good idea. Tsdoc generators can only do so much. I'll refresh my mind on it tomorrow

**Pull request merged**